### ISSUE TITLE
[derive] derive.param1 fails on `Fixpoint`

### ISSUE THREAD

**Issue by Columbus240** [LPCIC/coq-elpi#266]
I had a more complicated example, but rebuilt it like this. Interestingly, the error message changed. (Edit: it changed because I updated to the `master` branch)
```coq
Require Import List.
Import ListNotations.

From elpi.apps Require Import derive.

derive unit.
derive bool.
Elpi derive.param1 unit_eq.
derive list.
derive prod.
derive option.

Definition pair : Set := unit * unit.
Elpi derive.param1 pair.
Definition plist := list pair.
Elpi derive.param1 plist.

Fixpoint find_snd (pl : plist) (id : unit) : option unit :=
  match pl with
  | [] => None
  | (id0, p) :: tl =>
    if unit_eq id id0 then
      Some p
    else
      find_snd tl id
  end.
Fail Elpi derive.param1 find_snd.
(* This elpi program failed without giving a specific error message. Please report this inconvenience to the authors
of the program. *)

Definition find_snd' (pl : plist) (id : unit) : option unit :=
  option_map _ _ snd (find (fun p => unit_eq id (fst p)) pl).
Elpi derive.param1 option_map.
Elpi derive.param1 snd.
Elpi derive.param1 find.
Elpi derive.param1 fst.
Elpi derive.param1 find_snd'.
```
The second function is functionally equivalent, but doesn’t trigger the error.
I could minimize it further (given the above prelude) to:
```coq
Fixpoint foo (pl : plist) (id : unit) : option unit := None.
Fail Elpi derive.param1 foo.
```

**Comment by gares** [comment]
This is a bug, thanks for reporting. I did write a tentative fix in #269, but I had to change some ancient code I'm not sure to understand fully so I'll wait for some help by Cyril which actually wrote the original code.

**Comment by Columbus240** [comment]
Thanks for trying to fix it.
